src/query/storage/query_conversion_lru_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package storage
+
+import (
+	"testing"
+
+	"github.com/m3db/m3/src/m3ninx/idx"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLRU(t *testing.T) {
+	lru, err := NewQueryConversionLRU(5)
+	require.NoError(t, err)
+
+	// test add and get
+	lru.Add("a", idx.NewTermQuery([]byte("foo"), []byte("bar")))
+	lru.Add("b", idx.NewTermQuery([]byte("biz"), []byte("baz")))
+
+	var (
+		ok bool
+		q  idx.Query
+	)
+
+	q, ok = lru.Get("a")
+	require.True(t, ok)
+	assert.Equal(t, "term(foo, bar)", q.String())
+	q, ok = lru.Get("b")
+	require.True(t, ok)
+	assert.Equal(t, "term(biz, baz)", q.String())
+
+	// fill up the cache
+	lru.Add("c", idx.NewTermQuery([]byte("bar"), []byte("foo")))
+	lru.Add("d", idx.NewTermQuery([]byte("baz"), []byte("biz")))
+	lru.Add("e", idx.NewTermQuery([]byte("qux"), []byte("quz")))
+	lru.Add("f", idx.NewTermQuery([]byte("quz"), []byte("qux")))
```

Comment:
Can you check that the return types for the others are `false` and the return type for this one is `true`?